openedx/core/djangoapps/credit/partition_schemes.py

```diff
@@ -53,18 +54,47 @@ def get_group_for_user(cls, course_key, user, user_partition):
             string of allowed access group
         """
         checkpoint = user_partition.parameters['location']
+        # here getting cache key names for all models. So that we can make the
+        # list of keys and get the cache.get_many
```

Comment:
This function is getting a bit long.  I think it would be clearer if we moved the code that retrieves values into a helper method.